### PR TITLE
test: migrate path and paths to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/path.rs`
- `tests/testsuite/paths.rs`

### CODE REVIEW

File: tests/testsuite/path.rs [hunk 1]
```diff
@@ -73,44 +72,49 @@ fn cargo_compile_with_nested_deps_shorthand() {
...
 
     println!("cleaning");
     p.cargo("clean -v")
-        .with_stderr(
-            "[REMOVING] [CWD]/target\n\
-             [REMOVED] [..]",
-        )
+        .with_stderr_data(str![[r#"
+[REMOVING] [ROOT]/foo/target
+[REMOVED] [..] files, [FILE_SIZE]B total
```
Comment (eth3lbert) [inline tests/testsuite/path.rs:98]: This highlights the potential benefit of redacting the number of files, as suggested in https://github.com/rust-lang/cargo/pull/14096#discussion_r1644832897.

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
@bors r+

Thanks!

**Comment by weihanglo** [comment]
https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492637395#step:11:4033

You could add `edition = "2021"` to fix the test error.

**Comment by eth3lbert** [comment]
> https://github.com/rust-lang/cargo/actions/runs/9605300584/job/26492637395#step:11:4033
> 
> You could add `edition = "2021"` to fix the test error.

Addressed and rebased! Thanks for the review!

**Comment by eth3lbert** [comment]
Updated with auto-redaction applied!

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks!

@bors r+

**Pull request merged**